src/Components/Game/Game.js

```diff
@@ -1,111 +1,109 @@
+const Game = ({ get4RdmPokemon, fetchSinglePokemon}) => {
   const [winCounter, setWinCounter] = useState(0);
   const pokemons = get4RdmPokemon();
   const [pokemonChoices, setPokemonChoices] = useState(pokemons);
-  const [bestCounter, setBestCounter] = useState(0)
-  const wrongSoundObj = new Audio(wrongSound)
-  const rightSoundObj = new Audio(rightSound)
-  const [imageClassName, setImageClassName] = useState('single-pokemon')
+  const [bestCounter, setBestCounter] = useState(0);
+  const wrongSoundObj = new Audio(wrongSound);
+  const rightSoundObj = new Audio(rightSound);
+  const [imageClassName, setImageClassName] = useState('single-pokemon');
+  const [isWinner, setIsWinner] = useState(false);
+  const { singlePokemon, setSinglePokemon } = useContext(AppContext);
 
   const createOptionList = () => {
     return pokemonChoices.map(( pokemon, index ) => {
       return (
         <button
           name="user-button"
           onClick={(event) => {
-            checkForWin(event)
-            togglePokemonDisplay()
-            }
+            checkForWin(event);
+            togglePokemonDisplay();
+          }
           }
           id={`${pokemon.name}`}
           className="pokemon-button"
           key={index}
         >
           {pokemon.name}
         </button>
-      )
-    })
-  }
+      );
+    });
+  };
 
   const checkForWin = (event) => {
     const winner = singlePokemon.name;
 
-    if(winner === event.target.id && winCounter >= bestCounter) {
-      setWinCounter(winCounter + 1)
-      setBestCounter(winCounter + 1)
-      rightSoundObj.play()
+    if (winner === event.target.id && winCounter >= bestCounter) {
+      setWinCounter(winCounter + 1);
+      setBestCounter(winCounter + 1);
+      setIsWinner(true);
+      rightSoundObj.play();
     } else if (winner === event.target.id) {
-      setWinCounter(winCounter + 1)
+      setWinCounter(winCounter + 1);
+      setIsWinner(true);
+      rightSoundObj.play();
     } else {
-      wrongSoundObj.play()
-      setWinCounter(0)
+      wrongSoundObj.play();
+      setWinCounter(0);
+      setIsWinner(false);
     }
-  }
+  };
 
 
   const togglePokemonDisplay = () => {
-    setImageClassName('single-pokemon visible')
+    setImageClassName('single-pokemon visible');
     const turnDisplayOff = () => {
-      setImageClassName('single-pokemon')
-      setPokemonChoices(get4RdmPokemon())
-    }
-    setTimeout(turnDisplayOff, 2000)
-  }
+      setImageClassName('single-pokemon');
+      setPokemonChoices(get4RdmPokemon());
+    };
+    setTimeout(turnDisplayOff, 2000);
+  };
 
   useEffect(() => {
     const {url} = pokemonChoices[Math.floor(Math.random() * pokemonChoices.length)];
-
-    const fetchSinglePokemon = async () => {
-      try {
-        const singlePokemon = await getSinglePokemon(url)
-        setSinglePokemon(singlePokemon)
-      } catch (error) {
-        console.log(error);
-      }
-    }
-    fetchSinglePokemon();
-
+    fetchSinglePokemon(url);
```

Comment:
I think `fetchSinglePokemon` may change when the App re-renders? I'm not sure. So useEffect may think that `fetchSinglePokemon` is a prop that needs to be responded to. Does this need to be in a useEffect? Seems like you could just call `fetchSinglePokemon` after `togglePokemonDisplay` gets called.